### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by jared2501**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# gRPC/JSON Bridge
+
+A simple service that uses the [gRPC ServerReflection service](https://github.com/grpc/grpc/blob/master/src/proto/grpc/reflection/v1alpha/reflection.proto) to dynamically translate HTTP JSON requests into unary gRPC requests, that must have unary responses (streaming capabilities "coming soon").
```
Comment (uschi2000) [inline readme.md]: two sentences

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+message MyRequest {
+    string message = 1;
+}
+
+message MyResponse {
+    repeated uint32 array = 1;
+}
+```
+
+A curl request to the gRPC/JSON bridge such as:
```
Comment (uschi2000) [inline readme.md]: grammar

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+For example, given a service that implements the following gRPC service:
+
+```proto
+syntax = "proto3";
+
+package com.github.jared2501.foo;
+
+service ExampleService {
+    rpc ExampleMethod (TestMessage) returns (TestMessage);
```
Comment (uschi2000) [inline readme.md]: add doc

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  http://grpc-json-bridge.local:8014/example/com.github.jared2501.foo.ExampleService/ExampleMethod
+```
+
+Would yield a response of:
+
+```json
+{"array": [1, 2, 3]}
+```
+
+Note that the URL requested above follows the pattern:
```
Comment (uschi2000) [inline readme.md]: -note

File: readme.md (hunk 1 above)
Comment (uschi2000) [reply readme.md]: canonical JSON format + link

File: readme.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+- [grpc-ecosystem/grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway): requires annotating your gRPC proto definitions with special annotations and then running a tool which generates Go code that you must run inside a Go server.
+- [Envoy gRPC-JSON transcoder filter](https://www.envoyproxy.io/docs/envoy/latest/api-v1/http_filters/grpc_json_transcoder_filter): requires the gRPC proto definitions to be made available to Envoy.
+- [grpc-web](https://github.com/improbable-eng/grpc-web): An implementation of the [gRPC web protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) that removes the requirement for an HTTP/2 connection and allows JSON to be submitted as "application/grpc-web+json". However, the use of JSON requires the downstream server to support JSON marshalling (which the default generators do not produce).
+
+This repository offers a simple, quick solution that does not require changing your existing gRPC/Protobuf services and does not require the proto definitions to be made available statically to the server.
+
+## How To Build/Run
+
+For development purposes the server can be run by running:
```
Comment (uschi2000) [inline readme.md]: purposes,

File: readme.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+```bash
+./gradlew grpc-json-bridge-server:run
```
Comment (uschi2000) [inline readme.md]: + gd idea

File: readme.md (hunk 1 above)
Comment (jared2501) [reply readme.md]: done

File: readme.md (hunk 3 above)
Comment (jared2501) [reply readme.md]: done

File: readme.md (hunk 2 above)
Comment (jared2501) [reply readme.md]: done

File: readme.md (hunk 4 above)
Comment (jared2501) [reply readme.md]: done

File: readme.md (hunk 5 above)
Comment (jared2501) [reply readme.md]: done

File: readme.md (hunk 6 above)
Comment (jared2501) [reply readme.md]: done

**Pull request merged**